src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -371,12 +480,117 @@ extern "C" int32_t EnumCalendarInfo(
 			return EnumMonths(locale, calendarId, DateFormatSymbols::FORMAT, DateFormatSymbols::ABBREVIATED, callback, context);
 		case EraNames:
 		case AbbrevEraNames:
-			// NOTE: On Windows, the EraName is "A.D." and AbbrevEraName is "AD".
-			// But ICU/CLDR only supports "Anno Domini", "AD", and "A".
-			// So returning getEras (i.e. "AD") for both EraNames and AbbrevEraNames.
-			return EnumEraNames(locale, calendarId, callback, context);
+			return EnumEraNames(locale, calendarId, dataType, callback, context);
 		default:
 			assert(false);
 			return false;
 	}
 }
+
+/*
+Function:
+GetLatestJapaneseEra
+
+Gets the latest era in the Japanese calendar.
+*/
+extern "C" int32_t GetLatestJapaneseEra()
+{
+	UErrorCode err = U_ZERO_ERROR;
+	Locale japaneseLocale("ja_JP@calendar=japanese");
+	LocalPointer<Calendar> calendar(Calendar::createInstance(japaneseLocale, err));
+
+	if (U_FAILURE(err))
+		return 0;
+
+	return calendar->getMaximum(UCAL_ERA); 
+}
+
+/*
+Function:
+GetJapaneseEraInfo
+
+Gets the starting Gregorian date of the specified Japanese Era.
+*/
+extern "C" int32_t GetJapaneseEraStartDate(
+	int32_t era,
+	int32_t* startYear,
+	int32_t* startMonth,
+	int32_t* startDay)
+{
+	UErrorCode err = U_ZERO_ERROR;
+	Locale japaneseLocale("ja_JP@calendar=japanese");
+	LocalPointer<Calendar> calendar(Calendar::createInstance(japaneseLocale, err));
+	if (U_FAILURE(err))
+		return false;
+
+	calendar->set(UCAL_ERA, era);
+	calendar->set(UCAL_YEAR, 1);
+
+	// UCAL_EXTENDED_YEAR is the gregorian year for the JapaneseCalendar
+	*startYear = calendar->get(UCAL_EXTENDED_YEAR, err);
+	if (U_FAILURE(err))
+		return false;
+
+	// set the date to Jan 1
+	calendar->set(UCAL_MONTH, 0);
```

Comment:
we can have this clock part of the following loop if we need to simplify this code a little. something like init month and day as you did and the the loop will be something like 
while (era != currentEra)
{
   calculate the era with the month and day
  if current era!= calculated era then
      increment (day or month)
}